services/ymax-planner/test/plan-withdraw.property.test.ts

```diff
@@ -0,0 +1,214 @@
+      currentBalances,
+      delta,
+    );
+    const mergedTarget = { ...currentBalances, ...draftTarget };
+
+    const weights = Object.entries(currentBalances).map(
+      ([key, amount]) => [key, amount.value] as const,
+    );
+    const currentTotal = weights.reduce((acc, [, v]) => acc + v, 0n);
+    const expectedTotal = currentTotal + delta;
+    t.true(expectedTotal >= 0n);
+
+    const sumW = weights.reduce((acc, [, w]) => acc + w, 0n);
+    const shares = new Map(
+      weights.map(([key, weight]) => [key, (expectedTotal * weight) / sumW]),
+    );
+    const remainder =
+      expectedTotal -
+      [...shares.values()].reduce((acc, value) => acc + value, 0n);
+    const [maxKey] = weights.reduce<readonly [string, bigint]>(
+      (best, entry) => (entry[1] > best[1] ? entry : best),
+      ['<none>', -1n],
+    );
+
+    const totalAfter = Object.values(mergedTarget).reduce(
+      (acc, { value }) => acc + value,
+      0n,
+    );
+    t.is(totalAfter, expectedTotal);
+
+    for (const [key, share] of shares.entries()) {
+      const targetValue = mergedTarget[key]!.value;
+      const expectedValue = key === maxKey ? share + remainder : share;
+      t.is(targetValue, expectedValue);
+    }
+  },
+);
+
+const allKeys = ['Aave_Avalanche', 'Compound_Arbitrum', 'USDN'] as string[];
+
+testProp(
+  'computeWeightedTargets zeros current balances that lost their target allocation',
+  [
+    fc.record({
+      balances: fc.record({
+        Aave_Avalanche: fc.nat({ max: 5_000 }),
+        Compound_Arbitrum: fc.nat({ max: 5_000 }),
+        USDN: fc.nat({ max: 5_000 }),
+      }),
+      targetKeys: fc.subarray(allKeys, {
+        minLength: 1,
+        maxLength: allKeys.length - 1,
+      }),
+      withdrawal: fc.nat({ max: 15_000 }),
+    }),
+  ],
+  (t, { balances, targetKeys, withdrawal }) => {
+    const positiveBalances = Object.entries(balances).filter(([, v]) => v > 0);
+    const currentTotal = positiveBalances.reduce(
+      (sum, [, v]) => sum + BigInt(v),
+      0n,
+    );
+    fc.pre(currentTotal > 0n);
+    fc.pre(BigInt(withdrawal) <= currentTotal);
+
+    const missingKeys = allKeys.filter(key => !targetKeys.includes(key));
+    const missingWithValue = missingKeys.filter(key => balances[key] > 0);
+    fc.pre(missingWithValue.length > 0);
+
+    const targetAllocation = Object.fromEntries(
+      targetKeys.map((key, idx) => [key, BigInt(idx + 1)]),
+    );
+    const currentBalances = Object.fromEntries(
+      Object.entries(balances).map(([key, value]) => [
+        key,
+        makeDeposit(BigInt(value)),
+      ]),
+    );
+    const draft = computeWeightedTargets(
+      depositBrand,
+      currentBalances,
+      -BigInt(withdrawal),
+      targetAllocation,
+    );
+
+    for (const key of missingWithValue) {
+      t.true(key in draft);
+      t.is(draft[key]!.value, 0n);
+    }
+  },
+);
+
+test('planWithdrawFromAllocations zeros hubs and tops up cash before planning', async t => {
```

Comment:
"planWithdrawFromAllocations zeros hubs" is another good concept with a poor implementation.